debug.go

```diff
@@ -20,6 +20,12 @@ func (s *spanImpl) Lock() {
 }
 
 func (s *spanImpl) maybeAssertLocked() {
+	if s.tracer == nil {
+		s.Mutex.Unlock()
```

Comment:
Yeah, I considered that as well but with the absence of much code growth here I'm happier with the way it is.